distributed/scheduler.py

```diff
@@ -4138,6 +4193,52 @@ def decide_worker(self, ts):
 
         return worker
 
+    def transition_waiting_speculative(self, key):
+        try:
+
+            ts = self.tasks[key]
+
+            if self.validate:
+                # All dependencies are on the same worker
+                assert len({dts.processing_on for dts in ts.dependencies}) == 1
+
+            ws = list(ts.dependencies)[0].processing_on
+
+            worker = ws.address
+
+            duration = self.get_task_duration(ts)
+            # There are no comm costs if this is speculative
+            # comm = self.get_comm_cost(ts, ws)
+
+            ws.processing[ts] = duration
+            ts.processing_on = ws
+            ws.occupancy += duration
+            self.total_occupancy += duration
+            ts.state = "processing"
```

Comment:
Maybe the state should be "speculative" instead?  This would remove the need for the `.speculative` attribute, and maybe make some of the other handling in the scheduler more consistent.